local/blockchain.go

```diff
@@ -295,33 +298,39 @@ func (ln *localNetwork) setupWalletAndInstallSubnets(
 func (ln *localNetwork) installSubnets(
 	ctx context.Context,
 	numSubnets uint32,
-	baseWallet *refreshableWallet,
+	baseWallet primary.Wallet,
 	testKeyAddr ids.ShortID,
-) ([]ids.ID, error) {
+	pTXs []ids.ID,
+) (primary.Wallet, []ids.ID, error) {
 	println()
 	color.Outf("{{blue}}{{bold}}add subnets{{/}}\n")
 
 	subnetIDs, err := createSubnets(ctx, numSubnets, baseWallet, testKeyAddr)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	if numSubnets > 0 {
 		if err = ln.restartNodesWithWhitelistedSubnets(ctx, subnetIDs); err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 		println()
 		color.Outf("{{green}}reconnecting the wallet client after restart{{/}}\n")
 		clientURI, err := ln.getClientURI()
 		if err != nil {
-			return nil, err
+			return nil, nil, err
+		}
+		testKeychain := secp256k1fx.NewKeychain(genesis.EWOQKey)
+		allTxs := append(pTXs, subnetIDs...)
+		baseWallet, err = primary.NewWalletWithTxs(ctx, clientURI, testKeychain, allTxs...)
```

Comment:
is it possible to reuse setupWallet here?

Reply:
It is, but setupWallet is also doing more stuff like checking balance which seems unnecessary here